### ISSUE TITLE
Loops of logic checks

### ISSUE THREAD

**Issue by SilviaAmAm** [open-formulieren/open-forms#682]
When filling in a form in the SDK, the logic check sometimes get stuck in a loop. 
This can prevent the user from entering data or continuing to the next form step.

**Comment by sergei-maertens** [comment]
Also notice this with the appointments config, since it fetches the options for the dropdowns and the state is tracked in the Formio form - which of course gets replaced/re-rendered if the logic check triggers a React component re-render.

Just dropping some thoughts here to consider when we start working on solutions:

* keeping the appointment state/options in React state & pass them as options to the react-formio component (just like how we pass the `intl` option now). In the component(s), this is accessible via `this.options`.
* Check if we can modify components on the fly (with a hook maybe, `useInjectAppointmentConfig` before passing it to the underlying FormIO instance). This would stop us from doing state management in two different "domains" and having to deal with the aftermath of them affecting each other

**Comment by sergei-maertens** [comment]
The interaction with the appointment forms/components is quite problematic and a deep rabbit hole to go into.

Currently, the logic checks cause the dropdowns to "flicker", which is caused by updating the form configuration of the Formio.js form. We've tried working around it in various ways, to no avail. It's probably because of how the Formio webform rebuild works - all components in the tree are detached, re-initialized and re-attached. This has some DOM interaction, and adding/setting the dropdown values combined with the value results in a visual flicker.

The call to `setForm` is essential for the logic machinery. We also run into an issue where the backend uses Formio.js 4.13 component definitions, and the frontend has 4.12. This results in Formio's internal "change" detection to trigger a rebuild, even though the component definitions are functionality identical.

As an aside - the dropdowns are configured to refresh on changes of other components, which is not compatible with lazy-load. This in turn causes the dropdowns to fetch their options on step-load, with invalid query params. This is worked around for now by making the endpoint return an empty list of options rather than a HTTP 400.

Trying to prevent this call from happening in the first place by setting the component as disabled and enabling it with json logic (vanilla Formio functionality) introduced rendering options where the html5 widget is no longer used and leads us down another rabbit hole.

In short - a massive timesink without clear path forward. To be continued...

**Comment by joeribekker** [comment]
Check, removing from sprint for later.

**Comment by joeribekker** [comment]
Moved @sergei-maertens comments into #698 since the original loop bug was fixed.
